### ISSUE TITLE
ghc-lib-parser does not build with GHCJS: 'ghc_boot_platform.h' file not found

### ISSUE THREAD

**Issue by neongreen** [digital-asset/ghc-lib#204]
```
Preprocessing library for ghc-lib-parser-8.10.1.20200412..
Building library for ghc-lib-parser-8.10.1.20200412..

In file included from compiler/basicTypes/Avail.hs:7:0: error:

/nix/store/23r29nvkgr6xhr9xx3kbxqb313jxmans-ghcjs-8.6.0.1/lib/ghcjs-8.6.0.1/include/HsVersions.h:16:10: error:
     fatal error: 'ghc_boot_platform.h' file not found
   |
16 | #include "ghc_boot_platform.h"
   |          ^
#include "ghc_boot_platform.h"
         ^~~~~~~~~~~~~~~~~~~~~
1 error generated.
`cc' failed in phase `C pre-processor'. (Exit code: 1)
```

**Comment by neongreen** [comment]
I assume this file should be shipped with ghc-lib-parser, similarly to many other `.h` files, but I might be wrong.

**Comment by shayne-fletcher** [comment]
This file ships with `ghc-lib` versions for GHC flavor < 8.10. It is not produced for later versions (in this GHC commit it seems https://gitlab.haskell.org/ghc/ghc/commit/05419e55cab272ed39790695f448b311f22669f7).

**Comment by shayne-fletcher** [comment]
> `ghcjs-8.6.0.1/include/HsVersions.h:16:10: error`

ghc-lib ships with it's own `HsVersions.h`. Why are we pulling in the `ghcjs` one?

**Comment by shayne-fletcher** [comment]
The include dirs from `ghc-lib-parser.cabal` state:
```
    include-dirs:
        includes
        ghc-lib/stage0/lib
        ghc-lib/stage0/compiler/build
        compiler
```
and `compiler/HsVersions.h` is what we should be getting. I have no experience with ghcjs but I get the feeling somehow the include path is getting manipulated in an unexpected way.

**Comment by neongreen** [comment]
I renamed `HsVersions.h` along with references to it, to ensure it doesn't clash with GHC-shipped includes. That was enough for ghc-lib-parser to compile.

Perhaps it should be done for all includes, if there's an easy way to do so in the generator? It might also fix https://github.com/digital-asset/ghc-lib/issues/184:

```
Preprocessing library for ghc-lib-parser-8.8.0.20190424..
In file included from /usr/lib64/ghc-8.6.5/include/Stg.h:319,
```
